### PR TITLE
sped up vsEmit() of ihex and srec file formats

### PR DESCRIPTION
**Pull request by sprout42**
While testing parsing of large ihex and srec files identified that the `vsEmit()` functions for ihex and srec structures was inefficient.

### CODE REVIEW

File: vstruct/defs/ihex.py [hunk 1]
```diff
@@ -77,10 +77,7 @@ def vsEmit(self, fast=False):
             ffvals = [ ff.vsGetValue() for ff in self._vs_fastfields ]
             return struct.pack(self._vs_fastfmt, *ffvals)
 
-        ret = b''
-        for fname, fobj in self.vsGetFields():
-            ret += fobj.vsEmit() + b'\r\n'
-        return ret.decode('utf-8')
+        return b''.join(fobj.vsEmit() + b'\r\n' for _, fobj in self.vsGetFields())
```
Comment (williballenthin) [inline vstruct/defs/ihex.py:80]: `b'\r\n'.join(fobs.vsEmit() for _, fobj in ...)`

perhaps? tiny bit faster, tiny bit shorter.

Comment (sprout42) [reply vstruct/defs/ihex.py:80]: make sense to me, I'll update it

**Comment by sprout42** [comment]
So I'm not sure why the circleci job hasn't been run for this PR yet? If it's something I need to do on my end please let me know.

**Comment by atlas0fd00m** [comment]
i'm not certain, but it's possible you need to sign up for CircleCi on your github account?

**Comment by atlas0fd00m** [comment]
@rakuy0 any ideas on why CircleCI would not run on this PR?

**Review by rakuy0 — APPROVED** [review]

**Pull request merged**